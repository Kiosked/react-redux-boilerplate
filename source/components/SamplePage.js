import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import Layout from "./Layout.js";

const FormHeading = styled.h2`
    font-style: italic;
`;
const FormRow = styled.div`
    display: flex;
    flex-direction: row;
`;

class SamplePage extends Component {
    static propTypes = {
        inputValue: PropTypes.string.isRequired,
        onSetValue: PropTypes.func.isRequired
    };

    handleValueChange(event) {
        this.props.onSetValue(event.target.value);
    }

    render() {
        return (
            <Layout>
                <FormHeading>Test form</FormHeading>
                <FormRow>
                    <label>Type something:</label>
                    <input
                        type="text"
                        value={this.props.inputValue}
                        onChange={::this.handleValueChange}
                    />
                </FormRow>
            </Layout>
        );
    }
}

export default SamplePage;
