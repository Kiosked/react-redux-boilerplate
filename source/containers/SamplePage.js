import { connect } from "react-redux";
import SamplePage from "../components/SamplePage.js";
import { setSampleValue } from "../actions/sample.js";
import { getSampleValue } from "../selectors/sample.js";
export default connect(
    (state, ownProps) => ({
        inputValue: getSampleValue(state)
    }),
    {
        onSetValue: newValue => dispatch => {
            dispatch(setSampleValue(newValue));
        }
    }
)(SamplePage);
