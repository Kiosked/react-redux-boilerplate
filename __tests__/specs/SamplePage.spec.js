import React from "react";
import renderer from "react-test-renderer";
import SamplePage from "../../source/components/SamplePage.js";

const NOOP = () => {};

test("renders base component correctly", () => {
    const component = renderer.create(
        <SamplePage
            inputValue=""
            onSetValue={NOOP}
        />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
