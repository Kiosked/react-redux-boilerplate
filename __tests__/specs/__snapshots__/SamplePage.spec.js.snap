// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders base component correctly 1`] = `
<div
  className="sc-bdVaJa iaUkTG"
>
  <div
    className="sc-bwzfXH fJKfzp"
  >
    <header
      className="sc-htpNat ggFOZY"
    >
      <h1>
        Sample
      </h1>
    </header>
    <main
      className="sc-bxivhb kinwFJ"
    >
      <h2
        className="sc-ifAKCX cdZoGB"
      >
        Test form
      </h2>
      <div
        className="sc-EHOje eVWzyT"
      >
        <label>
          Type something:
        </label>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </main>
  </div>
</div>
`;
